### PR TITLE
feat: support client/server mode

### PR DESCRIPTION
**Pull request by knqyf263**
## Overview
Implement client/server mode. It's like the following:

```
$ trivy server -listen 0.0.0.0:8080
```

```
$ trivy client -remote http://localhost:8080 alpine:3.10
```

The protocol between a client and server is defined by Protocol Buffer. [Twirp](https://github.com/twitchtv/twirp) is used to generate Go code.

**This PR is a stacked PR.**

## This PR
Define services/messages for Protocol Buffer

## TODO
- [x] define a proto file (#295, this PR)
- [x] refactor (#296)
- [x] rpc
  - [x] client (#302)
  - [x] server (#303)
- [x] trivy client/server (#304)
- [x] integration test (#306)
- [x] add more tests (#307)

## Note
**Don't merge until the above PRs are merged**

### CODE REVIEW

File: rpc/detector/service.proto [hunk 1]
```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package trivy.detector;
+option go_package = "detector";
+
+service OSDetector {
+    rpc Detect(OSDetectRequest) returns (DetectResponse);
+}
+
+message OSDetectRequest{
```
Comment (danielpacak) [inline rpc/detector/service.proto]: Here and other places in `service.proto` you might want to reformat the file

File: rpc/detector/service.proto [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    string os_family = 1;
+    string os_name = 2;
+    repeated Package packages = 3;
+}
+
+message DetectResponse{
+    repeated DetectedVulnerability detected_vulnerabilities= 1;
```
Comment (danielpacak) [inline rpc/detector/service.proto]: I'd just call it `vulnerabilities`. The type is already defined in package `trivy.detector`

File: rpc/detector/service.proto [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    string file_path = 1;
+    repeated Library libraries = 2;
+}
+
+message Library{
+    string name = 1;
+    string version = 2;
+}
+
+message DetectedVulnerability {
```
Comment (danielpacak) [inline rpc/detector/service.proto]: It's in package `trivy.detector` so I'd remove the `Detected` prefix

File: rpc/detector/service.proto [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+message Package {
+    string name = 1;
+    string version = 2;
+    string release =3;
+    int32  epoch = 4;
+    string arch = 5;
+    string src_name = 6;
+    string src_version = 7;
+    string src_release = 8;
+    int32 src_epoch = 9;
```
Comment (danielpacak) [inline rpc/detector/service.proto]: What's the difference between epoch and src_epoch? Lets explain it in proto doc / comment

File: rpc/detector/service.proto [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+message DetectedVulnerability {
+    string vulnerability_id = 1;
+    string pkg_name = 2;
+    string installed_version = 3;
+    string fixed_version = 4;
+    string title = 5;
+    string description = 6;
+    string severity = 7;
```
Comment (danielpacak) [inline rpc/detector/service.proto]: What about defining enum?

File: rpc/detector/service.proto (hunk 1 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: I introduced a plugin of GoLand to format a proto file, but it appears to be insufficient...I will introduce clang-format. Thanks.

File: rpc/detector/service.proto (hunk 2 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: Do you suggest like the following?

```suggestion
    repeated DetectedVulnerability vulnerabilities= 1;
```

File: rpc/detector/service.proto (hunk 3 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: Fair enough.

File: rpc/detector/service.proto (hunk 1 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: I added two commands to Makefile.

File: rpc/detector/service.proto (hunk 4 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: I added a comment

File: rpc/detector/service.proto (hunk 5 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: 👍

File: rpc/detector/service.proto (hunk 2 above)
Comment (knqyf263) [reply rpc/detector/service.proto]: I fixed it

**Pull request merged**